src/audio/codec_adapter/codec/cadence.c

```diff
@@ -513,6 +513,12 @@ static int cadence_codec_process(struct comp_dev *dev)
 			return ret;
 	}
 
+	/* Proceed only if we have enough data to fill the input buffer completely */
+	if (codec->mpd.avail < codec->mpd.in_buff_size) {
+		comp_dbg(dev, "cadence_codec_process(): not enough data to start processing");
```

Comment:
this could be very noisy if period size < mpd.in_buff_size, also curious what the benefit is of moving this out of the core

Reply:
@cujomalainey the process API will be modifed to pass pointers to the input buffers and output buffers and their sizes. Eventually, the idea is to get rid of the in_buff_size in codec->mpd. The codec will have to use the passes buffer sizes when process API is invoked.

Reply:
and as for noise its not that different from today right? this debug message is already there in codec_adapter copy()